### PR TITLE
#FN:15 chore: Setting up Redux Tool Kit

### PR DESCRIPTION
**Pull request by Illustre13**
# Trello Board
- https://trello.com/c/MTktzQNx
# PR Description
In this PR I will be setting up the app to use Redux Tool Kit.
# Description of tasks that were expected to be completed
- [x] Created the redux store, the reducer, and the action to handle all possible state using `@reduxjs/toolkit`library
- [x] Created component to interact with the redux store
- [x] Rendered the component to our react app using `react-redux` library
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
To Test this PR
0. clone the project on `https://github.com/atlp-rwanda/ecommerce-app-cogito-fe.git`
2. run `yarn install` to install all required dependencies
3. run `yarn run dev`
# Screenshots (If appropriate/applicable)
- n/a
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: src/component/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { useSelector, useDispatch } from "react-redux";
+import { fetchIndexMessage } from "../redux/action/IndexAction";
+import { IndexState } from "../redux/store";
+
+function InitialMessage() {
+    const indexData = useSelector((state: IndexState) => state.index.data)
+    const indexStatus = useSelector((state: IndexState) => state.index.status)
+    const indexLoading = useSelector((state: IndexState) => state.index.loading)
+    const indexError = useSelector((state: IndexState) => state.index.error)  
+//   const index = useSelector(
```
Comment (Niyobuzima) [inline src/component/index.tsx]: why is this commented

**Review by Niyobuzima — CHANGES_REQUESTED** [review]
resolve the conflicts

File: src/component/index.tsx (hunk 1 above)
Comment (Illustre13) [reply src/component/index.tsx]: They are not necessary, I've removed them now.

**Review by uwdjany — APPROVED** [review]

**Review by Niyobuzima — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇 🥇 🥇

**Pull request merged**